packages/storage/src/deque_store.rs

```diff
@@ -352,7 +462,38 @@ where
         if self.start >= self.end {
             return None;
         }
-        let item = self.deque_store.get_at(self.storage, self.start);
+        let item;
+        match self.deque_store._get_offset_pos(self.storage, self.start) {
+            Ok(offset_pos) => {
+                let indexes_page = offset_pos / self.deque_store.page_size;
+                let index_pos = offset_pos % self.deque_store.page_size;
+                match self.saved_indexes.get(&indexes_page) {
+                    Some(indexes) => {
+                        if let Some(item_data) = indexes.get(&index_pos) {
+                            item = Ser::deserialize(item_data);
+                        } else {
+                            item = Err(StdError::generic_err("Item not found at this index."));
```

Comment:
Also here, I think "out of bounds" is a more appropriate message

Reply:
Again, I only want to return `out of bounds` error right after checking `pos >= len`